### PR TITLE
Add Vector.scalars

### PR DESCRIPTION
**Pull request by dabrahams**
This does not yet remove accesses to vectors through withUnsafeMutableBufferPointer, but now that scalars are required to be supported, we are in a position to remove that from the protocol. 

No change in performance.

### CODE REVIEW

File: Sources/SwiftFusion/Inference/ArrayBuffer+Vector.swift [hunk 1]
```diff
@@ -13,7 +13,97 @@
...
+    else {
+      self.init(endIn: c)
+    }
+  }
+  
+  internal init<C: Collection>(firstValidIn c: C)
+    where C.Index == OuterIndex, C.Element: Collection, C.Element.Index == InnerIndex
+  {
+    self.init(firstValidIn: c) { $0 }
+  }
```
Comment (marcrasi) [inline Sources/SwiftFusion/Inference/ArrayBuffer+Vector.swift:50]: This one is unused.

Comment (dabrahams) [inline Sources/SwiftFusion/Inference/ArrayBuffer+Vector.swift:50]:
```suggestion
```

Comment (dabrahams) [reply Sources/SwiftFusion/Inference/ArrayBuffer+Vector.swift]: I know; it all needs comments and tests, but this is a convenience for the case where you have a direct collection-of-collections, with no need for projection to get the inner collection.

**Comment by dabrahams** [comment]
Superseded by #165

**Pull request closed without merging**